packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { JsonMap } from '@salesforce/ts-types';
+import * as fs from 'fs';
+import { QueryResult } from 'jsforce';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { QUERY_RESULTS_DIR_PATH } from '../constants';
+import {
+  CsvDataProvider,
+  DataProvider,
+  JsonDataProvider
+} from './dataProviders';
+
+export enum FileFormat {
+  JSON = 'json',
+  CSV = 'csv'
+}
+
+const sfdxCoreExtension = vscode.extensions.getExtension(
+  'salesforce.salesforcedx-vscode-core'
+);
+const sfdxCoreExports = sfdxCoreExtension
+  ? sfdxCoreExtension.exports
+  : undefined;
+const { getRootWorkspacePath } = sfdxCoreExports;
+
+export class QueryDataFileService {
+  private dataProvider: DataProvider;
+
+  constructor(
+    private queryData: QueryResult<JsonMap>,
+    private format: FileFormat,
+    private documentName: string
+  ) {
+    this.dataProvider = this.getDataProvider();
+  }
+
+  protected getDataProvider(): DataProvider {
+    switch (this.format) {
+      case FileFormat.CSV:
+        return new CsvDataProvider(this.documentName);
+      case FileFormat.JSON:
+        return new JsonDataProvider(this.documentName);
+      default:
+        throw new Error('No DataProvider Found');
+    }
+  }
```

Comment:
@jgellin-sf I tried to do it the cool way as hash look-up but TS was fighting me on it so I moved on. The switch seems much simpler. I would love to get your help later maybe to see if I was on the right track for the alternative.